src/dmd/link.d

```diff
@@ -226,7 +226,120 @@ public int runLINK()
         if (phobosLibname)
             global.params.libfiles.push(phobosLibname.xarraydup.ptr);
 
-        if (target.mscoff)
+        if (target.omfobj)
```

Comment:
I would prefer if you just negated the condition and left the associated branches as they are to reduce unnecessary diff.